src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -72,6 +72,10 @@ public sealed class ConnectionString
         private TimeSpan? _connectTimeout;
         private string _databaseName;
         private bool? _fsync;
+        private bool _hasSslBeenSet;
```

Comment:
It shouldn't matter whether it was set implicitly because the scheme is `mongodb+srv` or explicitly in the connection string.

I don't think this field is needed.